Extension_AiCrawler_Markdown.php

```diff
@@ -402,4 +407,192 @@ public static function generate_markdown_on_save( $post_id, $post, $update ) { /
+	 *
+	 * @return void
+	 */
+	public static function flush_markdown_on_update( $post_id, $post_after, $post_before ) {
+		// Don't run on revisions.
+		if ( \wp_is_post_revision( $post_id ) ) {
+			return;
+		}
+
+		// If the slug changed, flush the old URL too.
+		if ( $post_before->post_name !== $post_after->post_name ) {
+			// Flush the llms.txt due to a change in post URL.
+			self::flush_llms_manifest();
+		}
+	}
+
+	/**
+	 * Flushes the markdown cache when the status of a post changes to non-public.
+	 *
+	 * This method is triggered on post status transitions and is used to clear
+	 * the markdown cache for the affected post when it transitions to a non-public status.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param string   $new_status The new status of the post.
+	 * @param string   $old_status The previous status of the post.
+	 * @param WP_Post  $post       The post object whose status has changed.
+	 *
+	 * @return void
+	 */
+	public static function flush_markdown_on_status_change( $new_status, $old_status, $post ) {
+		// Bail if no status change or old/new status isn't publish.
+		if (
+			$old_status === $new_status
+			|| (
+				'publish' !== $old_status
+				&& 'publish' !== $new_status
+			)
+		) {
+			return;
+		}
+
+		// Don't run on revisions.
+		if ( \wp_is_post_revision( $post->ID ) ) {
+			return;
+		}
+
+		$was_public = 'publish' === $old_status;
+		$is_public  = 'publish' === $new_status;
+
+		// If visibility affecting inclusion changed, refresh cache.
+		if ( $was_public && ! $is_public ) {
+			// Flush markdown cache URL for post.
+			self::flush_markdown_url_for_post( $post->ID );
+
+			// Flush the llms.txt due to post becoming private.
+			self::flush_llms_manifest();
+		} elseif ( ! $was_public && $is_public ) {
+			// Flush the llms.txt due to post moving to published.
+			self::flush_llms_manifest();
+		}
+	}
+
+	/**
+	 * Flush cached markdown URLs when a post is deleted.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param int     $post_id Post ID.
+	 * @param WP_Post $post    Post object provided by the action.
+	 *
+	 * @return void
+	 */
+	public static function flush_markdown_cache_on_delete( $post_id, $post = null ) {
+		if ( \wp_is_post_revision( $post_id ) ) {
+			return;
+		}
+
+		$status = \get_post_status( $post_id );
+
+		// Only flush/remove if it was publicly visible when trashed
+		if ( 'publish' === $status ) {
+			self::flush_markdown_url_for_post( $post_id );
+			self::flush_llms_manifest();
+		}
+	}
+
+	/**
+	 * Flush cached markdown URLs when a post is deleted.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param int    $post_id         Post ID.
+	 * @param string $previous_status Previous post status.
+	 *
+	 * @return void
+	 */
+	public static function flush_markdown_cache_on_restore( $post_id, $previous_status ) {
+		if ( \wp_is_post_revision( $post_id ) ) {
+			return;
+		}
+
+		// Only flush/remove if it was publicly visible when trashed
+		if ( 'publish' === $previous_status ) {
```

Comment:
The "untrashed_post" hook previous_state will be "draft" for posts and "inherit" for attachments.

"By default posts that are restored will be assigned a status of 'draft'."
See https://github.com/WordPress/wordpress-develop/blob/6.8.2/src/wp-includes/post.php#L4064

So, this method doesn't seem necessary, since the post will not be published on restore.